link/src/main/java/com/stripe/android/link/account/LinkAccountManagerImpl.kt

```diff
@@ -0,0 +1,289 @@
+package com.stripe.android.link.account
+
+import androidx.annotation.VisibleForTesting
+import com.stripe.android.core.Logger
+import com.stripe.android.core.exception.StripeException
+import com.stripe.android.link.BuildConfig
+import com.stripe.android.link.LinkConfiguration
+import com.stripe.android.link.LinkPaymentDetails
+import com.stripe.android.link.analytics.LinkEventsReporter
+import com.stripe.android.link.injection.LinkScope
+import com.stripe.android.link.model.AccountStatus
+import com.stripe.android.link.model.LinkAccount
+import com.stripe.android.link.repositories.LinkRepository
+import com.stripe.android.link.ui.inline.SignUpConsentAction
+import com.stripe.android.link.ui.inline.UserInput
+import com.stripe.android.model.ConsumerSession
+import com.stripe.android.model.ConsumerSessionLookup
+import com.stripe.android.model.ConsumerSignUpConsentAction
+import com.stripe.android.model.PaymentMethodCreateParams
+import com.stripe.android.payments.core.analytics.ErrorReporter
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.map
+import javax.inject.Inject
+
+/**
+ * Manages the Link account for the current user, persisting it across app usages.
+ */
+@LinkScope
+internal class LinkAccountManagerImpl @Inject constructor(
```

Comment:
nit: Can we just name this `DefaultLinkAccountManager`? This falls more in-line with how we name our base production implementations.